app/models/user_session.rb

```diff
@@ -60,4 +58,18 @@ def user_agent=(value)
     def set_unique_key
       self.key = SecureRandom.urlsafe_base64(32)
     end
+
+  private_class_method def self.user_session_ttl_parse(duration)
+    case duration
+    when NilClass, ""
+      2.weeks
+    when Numeric, /^\d+$/
+      ActiveSupport::Duration.seconds(duration.to_i)
+    else
+      raise ArgumentError, "user session TTL should be specified as a number of seconds"
+    end.freeze
```

Comment:
Hmmm do you want to have the object returned by the method frozen? I would not expect that.

Another thing instead of the `case .. when`

```ruby
def user_session_ttl_parse(duration)
  return 2.weeks if duration.blank?

  ActiveSupport::Duration.seconds(Integer(duration))
rescue ArgumentError
  raise ApplicationConfigurationError, "user session TTL should be specified as a number of seconds"
end
```
Creating a tagged class to know exactly where it belongs to